server.c

```diff
@@ -664,6 +664,22 @@ initialize_dname_compression_tables(struct nsd *nsd)
 	compressed_dname_offsets[0] = QHEADERSZ; /* The original query name */
 }
 
+static int
+set_cloexec(struct nsd_socket *sock)
+{
+	assert(sock != NULL);
+
+	if(fcntl(sock->s, F_SETFD, O_CLOEXEC) == -1) {
+		const char *socktype =
+			sock->addr.ai_family == SOCK_DGRAM ? "udp" : "tcp";
+		log_msg(LOG_ERR, "fcntl(..., O_CLOEXEC) failed for %s: %s",
+			socktype, strerror(errno));
+		return -1;
```

Comment:
The return value is not used, and -1 (instead of 0).

Reply:
In this case that's actually correct. It's a tri-state, not a boolean. The functions return `1` on success, `-1` on error and `0` if it's not implemented.

Reply:
Or, perhaps you meant the return value for `set_cloexec` itself?

Reply:
Yes the `set_cloexec` return value, you are remarking on `fcntl` above?

Reply:
In that case, it's actually correct. It behaves like the other socket option setters. I decided not to check the return value of `set_cloexec` in the function that opens the socket because it's not actually important yet, it's something eventually required for the CreDNS port.

Reply:
Okay, that is good to know.